libgnucash/app-utils/business-prefs.scm

```diff
@@ -125,6 +125,48 @@
     "g" (N_ "The default date format used for fancy printed dates.")
     #f))
 
+  ;; Accounting Period
+
+  (reg-option
+   (gnc:make-multichoice-callback-option
+    gnc:*book-accounting-period-label* "Accounting Period"
+    "a0" "Please choose the fiscal year period to be used in reports. \
+Please note this will affect the date specifiers used in most report \
+options. This will not affect the specialised tax reports which use \
+tailored dates."  #f
+    (list (vector #f
+                  "Custom selection"
+                  "Custom selection")
+          (vector '31-03
+                  "1-April to 31-March"
+                  "1-April to 31-March")
+          (vector '05-04
+                  "6-April to 5-April"
+                  "6-April to 5-April")
+          (vector '30-06
+                  "1-July to 30-June"
+                  "1-July to 30-June")
+          (vector '30-09
+                  "1-October to 30-September"
+                  "1-October to 30-September")
+          (vector '31-12
+                  "1-January to 31-December"
+                  "1-January to 31-December"))
+    #f
+    (lambda (x)
+      (gnc-option-db-set-option-selectable-by-name
+       options gnc:*book-accounting-period-label* "Custom End of Financial Year"
+       (not x)))))
+
+  (reg-option
+   (gnc:make-date-option
+    gnc:*book-accounting-period-label* "Custom End of Financial Year"
+    "a2"
+    "Please choose the custom end of financial year. Please be aware \
+only the day and month are used. The year stated is unused."
+    (lambda () (cons 'absolute (current-time)))
+    #f 'absolute #f))
+
```

Comment:
With this commit the user now has two ways to specify an accounting period:

- via book options in File->Properties
- user-wide via Edit->Preferences

IMO this is a good recipe for confusion.

I do prefer the book options method. It's how it should have been implemented in the first place. However I think that means we need to migrate away from the user-wide option. Which in turn means backwards incompatible changes, so this should go on master to be handled properly.